homeassistant/components/weatherflow_cloud/sensor.py

```diff
@@ -176,35 +302,161 @@ async def async_setup_entry(
+
+    def _generate_unique_id(self) -> str:
+        """Generate a unique ID for the sensor."""
+        if self.device_id is not None:
+            return f"{self.station_id}_{self.device_id}_{self.entity_description.key}"
+        return f"{self.station_id}_{self.entity_description.key}"
+
+    @property
+    @abstractmethod
+    def native_value(self) -> StateType | date | datetime | Decimal:
+        """Abstract method for native value."""
+
+    @property
+    def available(self) -> bool:
+        """Get if available."""
+
+        if not super().available:
+            return False
+
+        if self.device_id is not None:
+            # Websocket sensors - have Device IDs
+            return bool(
+                self.coordinator.data
+                and self.coordinator.data[self.station_id][self.device_id] is not None
+            )
+
+        # Check availability of a REST based sensor
+        return bool(self.coordinator.data)
+
+
+class WeatherFlowWebsocketSensorObservation(WeatherFlowSensorBase):
+    """Class for Websocket Observations."""
+
+    entity_description: WeatherFlowCloudSensorEntityDescriptionWebsocketObservation
+
+    @property
+    def native_value(self) -> StateType | date | datetime | Decimal:
+        """Return the native value."""
+        if self.coordinator.data and self.device_id is not None:
+            data = self.coordinator.data[self.station_id][self.device_id]
+            return self.entity_description.value_fn(data)
+        return None
+
+
+class WeatherFlowWebsocketSensorWind(WeatherFlowSensorBase):
+    """Class for wind over websockets."""
+
+    entity_description: WeatherFlowCloudSensorEntityDescriptionWebsocketWind
+
+    @property
+    def native_value(self) -> StateType | datetime:
+        """Return the native value."""
+
+        if self.coordinator.data and self.device_id is not None:
+            data = self.coordinator.data[self.station_id][self.device_id]
+            return self.entity_description.value_fn(data)
+
+        return None
+
+    @property
+    def icon(self) -> str | None:
+        """Get icon."""
+
+        value = (
+            int(self.native_value)
+            if self.native_value is not None
+            and isinstance(self.native_value, (int, float, str))
+            else None
+        )
+
+        if value and self.entity_description.icon_fn is not None:
+            return self.entity_description.icon_fn(value)
+        return None
+
+
+class WeatherFlowCloudSensorREST(WeatherFlowSensorBase):
+    """Class for a REST based sensor."""
+
+    entity_description: WeatherFlowCloudSensorEntityDescription
+
+    coordinator: WeatherFlowCloudUpdateCoordinatorREST
 
     @property
     def native_value(self) -> StateType | datetime:
-        """Return the state of the sensor."""
-        return self.entity_description.value_fn(self.station.observation.obs[0])
+        """Return the native value."""
+        if self.coordinator.data:
```

Comment:
idem

Reply:
This makes sense since we already hit the REST sensor at startup... I was being extra protective